vllm/core/scheduler.py

```diff
@@ -1333,4 +1333,7 @@ def _get_num_new_tokens(self, seq_group: SequenceGroup,
         if enable_chunking and len(seqs) == 1:
             num_new_tokens = min(num_new_tokens,
                                  budget.remaining_token_budget())
+        if not enable_chunking and status == SequenceStatus.RUNNING:
+            num_new_tokens = min(num_new_tokens,
+                                 budget.remaining_token_budget())
```

Comment:
I think this has a bug if the sequence is running with beam search (`len(seqs)` > 1 and status=running --> number of scheduled decode tokens is inconsistent with budget).